docs/functional/architecture/0001-Field_View_Frontend_Design.md

```diff
@@ -87,7 +84,38 @@ This step makes use of the type information on expressions, for example to decid
+
+#### PAST parser
+Parses from Python AST into PAST.
+
+#### PAST Lowering
+Lowers from PAST to Iterator IR.
+
+__Field operator out argument slicing__
+The lowering contains some complex logic to emulate slicing of fields in order to restrict the output domain of a field operator call. This contradics guiding principle (3) and should be removed in the future. The concept of specifying the domain for Field operators was not investigated during the frontend design as we concentrated on the beautified iterator dialect where the output domain is explicitly given when lifting a local operator to a field operator. The alternative to also explicitly specify the domain for calls to field operators was rejected as it:
+- would require introducing significantly more syntax in order to specify domains, which was not only infeasible to implement in time, but there also didn't exist an accepted syntax to do so.
+- no intuitive syntax was found to do this
+  - `field_operator(in_field, out=out_field, domain=output_domain)`: Possible, but 
+  - `field_operator[output_domain](in_field, out=out_field)` This syntax is already used to create a field operator from a local operator.
+
+### Decorators (@program, @field_operator)
+
+#### What to keep
+The decorators are designed to be constructed from the root node of their respective dialect contrary to a python function object. This decision has been made to allow programmatically generating instances thereof.
+
+The Program decorator is not aware of the concept of a FieldOperator but instead uses a generic mechanism (`GTCallable`) to inject functions into the resulting ITIR. This allows to keep the the various decorators independent from each other and avoids coupling the typing system (which is conceptually on a lower level), to them (e.g. `make_symbol_type_from_value` uses the `__gt_type__` method).
+
 #### What could be changed
-##### Ad-hoc pass management
-Since this lowering goes from a statement based AST to a functional IR, all the statements in the AST have to be inlined. This happens in one of the visitor methods currently which calls a FOAST pass, which is perhaps not ideal.
 
+The GTCallable interface it no stable and was born as an ad-hoc concept to fullfil the requirements noted above. In particular the fact that it mixes types, i.e. a concept only found in the frontend, with the untyped iterator IR is not ideal. This leads to the strange situation, where the result of executing the ITIR, i.e. a stencil call, of a Field operator is a value, while the type suggests a Field return value and a call to the decorated function returns nothing.
```

Comment:
typos: `it no stable` -> `is not stable`, `fullfil` -> `fulfill`. Probably should run a spellchecker.